From fe2d69028796aafa23d983d59c59903ebde5b855 Mon Sep 17 00:00:00 2001
From: Harshitha Prem <quic_hprem@quicinc.com>
Date: Fri, 30 Sep 2022 10:54:29 +0530
Subject: [PATCH] hostapd: 6GHz channels are not sent for acs scan

In single wiphy, when 6GHz band is configured as auto channel,
hostapd does not send 6GHz channel for scan instead it sends
5GHz channel because of incorrect mode selected during init.

Hence, while selecting the HW mode along with mode check,
it has to check the configured operating class
and frequency.

Signed-off-by: Harshitha Prem <quic_hprem@quicinc.com>
---
 src/ap/hw_features.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

--- a/src/ap/hw_features.c
+++ b/src/ap/hw_features.c
@@ -1213,8 +1213,13 @@ int hostapd_select_hw_mode(struct hostap
 			    !hw_mode_get_channel(mode, iface->freq, &chan))
 				continue;
 
-			iface->current_mode = mode;
-			break;
+			if ((is_6ghz_op_class(iface->conf->op_class) &&
+			     is_6ghz_freq(mode->channels[0].freq)) ||
+			    (!is_6ghz_op_class(iface->conf->op_class) &&
+			     !is_6ghz_freq(mode->channels[0].freq))) {
+				iface->current_mode = mode;
+				break;
+			}
 		}
 	}
 
